src/store/preferences.ts

```diff
@@ -150,17 +180,28 @@ export const actions = {
           Authorization:  `Basic ${ window.btoa(`${ user }:${ password }`) }`,
           'Content-Type': 'application/x-www-form-urlencoded',
         }),
-        body: JSON.stringify(state.preferences),
+        body: JSON.stringify(proposal),
       });
 
+    if (!result.ok) {
+      const severities = { ...state.severities, error: true };
+
+      commit('SET_SEVERITIES', severities);
+      commit('SET_PREFERENCES_ERROR', await result.text());
```

Comment:
I wonder if we should make that endpoint return JSON (or a code path for `Accept: application/json`).

Reply:
That would be preferable imo 😄